src/realm/table.hpp

```diff
@@ -402,6 +403,9 @@ class Table {
     std::optional<Mixed> min(ColKey col_key, ObjKey* = nullptr) const;
     std::optional<Mixed> max(ColKey col_key, ObjKey* = nullptr) const;
     std::optional<Mixed> avg(ColKey col_key, size_t* value_count = nullptr) const;
+    
+    // Calculate the distance between the two vectors (embeddings)
+    float dist_knn(const std::vector<float>& query_data, ColKey column, ObjKey row, hnswlib::SpaceInterface<float>& s) const;
```

Comment:
This should not be an operation on Table. Could just be a lambda within `SemanticSearchDescriptor::execute`